lib/node_modules/@stdlib/math/base/special/logaddexp/test/test.js

```diff
@@ -0,0 +1,113 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var abs = require( '@stdlib/math/base/special/abs' );
+var PINF = require( '@stdlib/constants/math/float64-pinf' );
+var NINF = require( '@stdlib/constants/math/float64-ninf' );
+var EPS = require( '@stdlib/constants/math/float64-eps' );
+var randu = require( '@stdlib/random/base/randu' );
+var logaddexp = require( './../lib' );
+
+
+// FIXTURES //
+
+var data = require( './fixtures/python/data.json' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.equal( typeof logaddexp, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns `NaN` if provided `NaN`', function test( t ) {
+	var v;
+
+	v = logaddexp( NaN, 1 );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	v = logaddexp( 1, NaN );
+	t.equal( isnan( v ), true, 'returns NaN' );
+	t.end();
+});
+
+tape( 'the function returns `+infinity` if provided with `+infinity` and `x`', function test( t ) {
+	var iterations;
+	var x;
+	var v;
+	var i;
+
+	iterations = 1000;
+	for ( i = 0; i < iterations; i++ ) {
+		x = ( randu() * 1000.0 ) - 500.0;
+		v = logaddexp( PINF, x );
+		t.strictEqual( v, PINF, 'returns +infinity when provided +infinity and '+x+'.' );
+		v = logaddexp( x, PINF );
+		t.strictEqual( v, PINF, 'returns +infinity when provided +infinity and '+x+'.' );
+	}
+	t.end();
+});
+
+tape( 'the function returns `x` if provided with `-infinity` and `x`', function test( t ) {
```

Comment:
Similar comment as L56.

Reply:
I just thought about it and is it too vague to have `the function returns x if provided with -infinity` because it is not exactly clear what `x` is in the scenario?

And now that I'm typing this out, I thought about logaddexp(-infinity, -infinity) which I assume should be 0.

I figure another test for that case would be good.

Reply:
Ah, sorry. I did not think clearly about this. Yes, you are correct regarding ambiguity. The test description can stay as is.